client_test.go

```diff
@@ -34,6 +34,21 @@ func TestDialUDP(t *testing.T) {
 	}
 }
 
+func TestDialTimeoutTCP(t *testing.T) {
+	m := new(Msg)
+	m.SetQuestion("miek.nl.", TypeSOA)
+
+	// insomniac.slackware.it is configured to drop connections to 5053 for this
+	// test to succeed
+	conn, err := DialTimeout("tcp", "insomniac.slackware.it:5053", time.Millisecond*100)
```

Comment:
How to do get to the time out?

Not to sure this test adds anything

Reply:
it tries to connect to that host:port that is configured with a DROP rule on netfilter, and this will trigger the connect timeout. I can clarify the comment if necessary

Reply:
Additionally I would configure the same DROP rule miek.nl rather than on my own host since you already use miek.nl's server for some tests. This would help keeping the tests more consistent, and leave you in control of where it connects to

Reply:
yeah.. I know :) Also not perfect. We might want push this to something more official, but $$$. We should be conservative in adding remote endpoints that require fiddling and make (more) tests depend on that.

Reply:
I totally agree with you. On the short term: are you OK to move this test to point to a port on miek.nl configured to drop connections?

On the long term: what other provider would you see fit? And how much money would it cost? This is a very useful and used library, I am pretty sure it wouldn't be hard to raise some money or sponsorship to have a better infra (and I could contribute as well)

Reply:
Doing it on miek.nl is just as bad - the only reason that I originally did it was because I saw no other option and I was less likely to leave the project.

We should drop this PR and open an issue to discuss what we need from an external test server, how we pay for it and how we provision it.

Reply:
Agreed. I am thinking of other options for this specific test case, and thought that if we can assign a new local IP (e.g. `127.0.0.3` or similar) and set a DROP rule on that IP:port this test can be still valid. I can update the PR and see how does Travis-CI behave